Added environment platform config

andrii-ivaniuk opened the pull request.

stijn-janssens commented on backend/app/environments/schema.py:
```diff
@@ -18,27 +17,43 @@ class GetEnvironment(Environment):
 
-class Config(RootModel[Dict[str, _AWSS3Config]]):
-    pass
+class _AWSGlueConfig(ORMModel):
```
I would put all of these configs in separate files

stijn-janssens replied:
similar to my approach in data outputs branch

andrii-ivaniuk replied:
How do you suggest splitting it? Based on which entity?

stijn-janssens replied:
Have a file per service. I have S3DataOutput and GlueDataOutput separated, but inheriting from a generic DataOutput.
Same would go for these configs.

andrii-ivaniuk replied:
Fixed

stijn-janssens commented on backend/app/environments/schema.py:
```diff
@@ -18,27 +17,43 @@ class GetEnvironment(Environment):
 
 class EnvPlatformServiceConfig(ORMModel):
-    config: Config
+    config: list[_AWSS3Config | _AWSGlueConfig]
```
will we save this in the database as a single entry per environment and platform? Or will we also split per service?
Here you chose to do it as a single entry, why did you choose that and not the other option?

andrii-ivaniuk replied:
It is env,  platform, and service granularity. 

class EnvPlatformServiceConfig(Base, BaseORM):
    __tablename__ = "env_platform_service_configs"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    environment_id: Mapped[UUID] = mapped_column(ForeignKey("environments.id"))
    platform_id: Mapped[UUID] = mapped_column(ForeignKey("platforms.id"))
    service_id: Mapped[UUID] = mapped_column(ForeignKey("platform_services.id"))
    config = Column(String)

andrii-ivaniuk replied:
It is a list because it can be multiple identifiers fo e.g. s3

stijn-janssens commented on backend/app/environments/schema.py:
```diff
@@ -18,27 +17,43 @@ class GetEnvironment(Environment):
+    can_read_from: list[str]
+
+
+class EnvPlatformConfig(ORMModel):
```
I would also split up the platform and platform service configs. This feels like a god file now

andrii-ivaniuk replied:
Why do you mean under `platform` and `platform service configs`?

stijn-janssens reviewed (commented):
I think there are still things wrong with the mapping between your Pydantic schemas and the ORM model in SQL Alchemy. Up till now, these mappings match perfectly.
Why did you not include things like platform_id etc in the pydantic schemas?

andrii-ivaniuk commented:
> I think there are still things wrong with the mapping between your Pydantic schemas and the ORM model in SQL Alchemy. Up till now, these mappings match perfectly. Why did you not include things like platform_id etc in the pydantic schemas?

Do we need it? If yes, I'll add it.

andrii-ivaniuk commented:
> I think there are still things wrong with the mapping between your Pydantic schemas and the ORM model in SQL Alchemy. Up till now, these mappings match perfectly. Why did you not include things like platform_id etc in the pydantic schemas?

Added metadata about environment, platform, and service. Added endpoint that returns all the configs for a specific environment.
